### PR TITLE
Use explicit imports

### PR DESCRIPTION
**Pull request by nstarman**
Changing the imports to be explicit. I prefer it, and so does my static checker.
While more verbose, I also think this is much more explicit and now I can tell at a glance what is imported.
Also, it's going to make some planned deprecations easier.
There shouldn't be any API change in this PR.

### CODE REVIEW

**Comment by nstarman** [comment]
Requires #14979.

**Review by eerovaher — COMMENTED** [review]
I don't mind making the imports more explicit, but I do have a couple of questions about what this pull request is doing exactly.

File: astropy/cosmology/__init__.py [hunk 1]
```diff
@@ -8,24 +8,56 @@
...
+    "cosmology_equal",
+    # Parameter
+    "Parameter",
+    # Realizations
+    "available",
+    "default_cosmology",
+    # Units
+    "units",
+]
+__all__ += realizations.__all__  # Add the built-in cosmo realizations to __all__
```
[the remark this answers is not in the document]
Comment (nstarman) [reply astropy/cosmology/__init__.py:60]: Yes. I had a followup PR for this, but I can do it in this one.

File: astropy/cosmology/core.py [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 from __future__ import annotations
 
+__all__ = ["Cosmology", "CosmologyError", "FlatCosmologyMixin"]
```
[the remark this answers is not in the document]
Comment (nstarman) [reply astropy/cosmology/core.py:5]: PEP8. It was recently pointed out to me that PEP8 says `__all__` should precede all imports except `__future__`

[the remark this answers is not in the document]
Comment (nstarman) [reply astropy/cosmology/core.py:5]: I'll do this in a followup PR, cleaning up the whole subpackage for `__all__` and the other module-level dunders (which should also be moved).

[the remark this answers is not in the document]
Comment (nstarman) [reply astropy/cosmology/core.py]: Done!

**Review by eerovaher — APPROVED** [review]
`__all__` is now explicitly written out in all of `astropy.cosmology`, so this can be merged.

**Pull request merged**